src/metrics.rs

```diff
@@ -141,13 +186,20 @@ impl<'a> Visit for MetricVisitor<'a> {
     }
 
     fn record_u64(&mut self, field: &Field, value: u64) {
+        #[cfg(feature="otel_unstable")]
+        if let Some(metric_name) = field.name().strip_prefix(METRIC_PREFIX_GAUGE) {
+            self.visited_metrics
+                .push((metric_name, InstrumentType::GaugeU64(value)));
+            return;
+        }
         if let Some(metric_name) = field.name().strip_prefix(METRIC_PREFIX_MONOTONIC_COUNTER) {
             self.visited_metrics
                 .push((metric_name, InstrumentType::CounterU64(value)));
-        } else if let Some(metric_name) = field.name().strip_prefix(METRIC_PREFIX_COUNTER) {
+            return;
+        }
```

Comment:
Not sure why we should drop the `else if` here?

Reply:
I guess it was done s because there is no way to guard `else if let` by feature gate. I will rollback `else if` as it was, and will add additional gated `if let` at the end of function

Reply:
[fixed](https://github.com/tokio-rs/tracing-opentelemetry/pull/129/commits/d2d0ed2fc3aa88339ce0e722795ca608a92c3102)